dev-libs/libbase58/libbase58-0.1.4-r1.ebuild

```diff
@@ -0,0 +1,48 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+inherit autotools multilib-minimal
+
+DESCRIPTION="C implementation of Bitcoin's base58 encoding"
+HOMEPAGE="https://github.com/luke-jr/libbase58"
+LICENSE="MIT"
+
+SRC_URI="https://github.com/luke-jr/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+SLOT="0/0"
+KEYWORDS="~amd64 ~arm ~mips ~ppc ~ppc64 ~x86"
+IUSE="test tools"
+RESTRICT="!test? ( test )"
+
+# NOTE: If not testing, we don't need non-native libgcrypt
+RDEPEND="tools? ( dev-libs/libgcrypt )"
+DEPEND="${RDEPEND}
+	test? ( dev-libs/libgcrypt[${MULTILIB_USEDEP}] )
+"
+
+src_prepare() {
+	eapply_user
+	eautoreconf
+
+	# NOTE: Needed because test suite uses srcdir instead of builddir to set PATH
+	multilib_copy_sources
+}
+
+multilib_src_configure() {
+	local myeconf=(
+		--disable-static
+		LIBGCRYPT_CONFIG="${EPREFIX}/usr/bin/${CHOST}-libgcrypt-config"
+	)
+	if multilib_is_native_abi && use tools; then
+		myeconf+=( --enable-tool )
+	elif use test; then
+		myeconf+=( --enable-tool --bindir='/TRASH' )
+	else
+		myeconf+=( --disable-tool )
+	fi
+	econf "${myeconf[@]}"
+}
+
+multilib_src_install_all() {
+	rm -rf "${ED}/TRASH" || die
+}
```

Comment:
I'm going to add 
```
multilib_src_install_all() {                                                                        
    if use test; then                                                                               
        rm -r "${ED}/TRASH" || die                                                                  
    fi                                                                                              
}                                                                                                   
```
which seems to work and do the job. Rest looks good, thanks!

Reply:
That WON'T work... TRASH won't exist with USE `tools` and only one target ABI.

Reply:
For some reason it doesn't error out with `USE="tools"` but I see where the logic is coming from. It does fail with `ABI_X86="(64) -32 (-x32)"`. I'll fix it in a moment, need to do few test runs. Thanks for pinging!

Reply:
Ok, 47b6f9481bd5f547af2134a5dea2e15bd45ca889 I honestly have **no** idea why we didn't just go with the `-f`. Please change it like that on the next version bump. I'm too embarrased to make a commit straight after...

Reply:
Okay the final version: 
```
multilib_src_install_all() {
	if use test; then
		# It's hard to control this directory with multilib_is_native_abi && use tools, hence -f.
		rm -rf "${ED}/TRASH" || die
	fi
}
```
Should be good. I needed to leave 15 minutes ago so my brain doesn't bend to this anymore. This should however work, or not work, in every case. But not die while emerging.